waspc/data/Generator/templates/sdk/wasp/server/auth/hooks.ts

```diff
@@ -0,0 +1,42 @@
+import type { Request as ExpressRequest } from 'express'
+import type { ProviderId, createUser } from '../../auth/utils.js'
+import { prisma } from '../index.js'
+
+type CommonInput = {
+  hookName: string
+  prisma: typeof prisma
+  req: ExpressRequest
+}
+
+/* On Before Signup Hook */
+export type OnBeforeSignupHookFn = (
+  params: OnBeforeSignupHookFnInput,
+) => Promise<void>
+
+type OnBeforeSignupHookFnInput = {
+  providerId: ProviderId
+} & CommonInput
+
+/* On After Signup Hook */
+export type OnAfterSignupHookFn = (
+  params: OnAfterSignupHookFnInput,
+) => Promise<void>
```

Comment:
What happens when a user wants the hook to be a non-async function.

Reply:
Yep, we need to account for that 👍

Reply:
I've adjusted the return type to allow for async and sync functions.